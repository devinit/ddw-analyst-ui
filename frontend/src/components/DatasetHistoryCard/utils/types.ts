export interface SavedQueryData {
  id: number;
  saved_query_db_table: string;
  operation: number;
  full_query: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
}
