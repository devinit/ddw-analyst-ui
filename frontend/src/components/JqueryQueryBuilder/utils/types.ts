export type JqueryQueryBuilderFilter = {
  condition?: string;
  rules?: (JqueryQueryBuilderFilterComparator | JqueryQueryBuilderFilter)[];
};

export type JqueryQueryBuilderHaving = {
  condition?: string;
  rules?: (JqueryQueryBuilderHavingComparator | JqueryQueryBuilderHaving)[];
};

export type JqueryQueryBuilderHavingComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value?: string | number | { plain: string | number } | { column: string; aggregate: string };
  values?: { value: string; label: string }[];
};

export type JqueryQueryBuilderFilterComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value: string | number;
};
export type JqueryQueryBuilderComps =
  | 'equal'
  | 'not_equal'
  | 'less'
  | 'less_or_equal'
  | 'greater'
  | 'greater_or_equal'
  | 'between'
  | 'in';

export type JqueryQueryBuilderFieldData = {
  id: string | number;
  label: string;
  type: string;
};

export type JqueryQueryBuilderIcons = {
  add_group: string;
  add_rule: string;
  remove_group: string;
  remove_rule: string;
  error: string;
};
