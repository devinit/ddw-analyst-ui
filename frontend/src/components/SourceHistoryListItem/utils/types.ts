export interface FrozenData {
  id: number;
  parent_db_table: string;
  frozen_db_table: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
  logs: string | null;
}
