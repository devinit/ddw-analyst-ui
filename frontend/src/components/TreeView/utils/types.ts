import { ReactElement } from 'react';

export interface TreeViewProps {
  data: Data[];
  depth?: number;

  deleteElement?: ReactElement;

  getStyleClassCb?: (node: Data) => void;

  isCheckable?: (node: Data, depth?: number) => void;
  isDeletable?: (node: Data, depth?: number) => void;
  isExpandable?: (node: Data, depth?: number) => void;

  keywordChildren?: string;
  keywordChildrenLoading?: string;
  keywordKey?: string;
  keywordLabel?: string;

  loadingElement?: ReactElement;
  noChildrenAvailableMessage?: string;

  onCheckToggleCb?: (data: Data[], depth?: number) => void;
  onDeleteCb?: (node: Data, data: Data[], depth?: number) => void;
  onExpandToggleCb?: (node: Data, depth?: number) => void;
  onUpdateCb?: (data: unknown[], depth?: number) => void;

  transitionEnterTimeout?: number;
  transitionExitTimeout?: number;
}

export interface Data {
  id: number;
  name: string;
  isExpanded?: boolean;
  isChecked?: boolean;
  isChildrenLoading?: boolean;
  children?: Data[];
}
