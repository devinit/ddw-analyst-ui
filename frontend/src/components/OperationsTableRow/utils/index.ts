import axios from 'axios';
import * as localForage from 'localforage';
import { api, localForageKeys } from '../../../utils';

interface ServerResponse {
  id: number;
  indicator: string;
  indicator_acronym: string;
  source: string;
}

interface SourcesResponse {
  status: number;
  statusText: string;
  data: ServerResponse;
}

export const fetchSource = async (sourceId: number): Promise<SourcesResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);

  return await axios.request({
    url: api.routes.SINGLE_SOURCE.replace('{sourceId}', `${sourceId}`),
    method: 'get',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
  });
};
