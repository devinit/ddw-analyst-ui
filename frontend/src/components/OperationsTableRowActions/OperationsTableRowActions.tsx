import * as React from 'react';

export const OperationsTableRowActions: React.SFC = props => {
  return (
    <React.Fragment>
      { props.children }
    </React.Fragment>
  );
};
