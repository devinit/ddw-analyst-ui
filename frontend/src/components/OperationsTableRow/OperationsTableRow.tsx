import React, { FunctionComponent } from 'react';
import { DatasetRowItem } from '../DatasetRowItem';
import { OperationsTableRowActions } from '../OperationsTableRowActions';

export interface OperationsTableRowProps {
  count: number;
  name: string;
  updatedOn: string;
  isDraft: boolean;
  classNames?: string;
}

export const OperationsTableRow: FunctionComponent<OperationsTableRowProps> = (props) => {
  const renderActions = (): React.ReactNode =>
    React.Children.map(props.children, (child) => {
      if (React.isValidElement(child) && child.type === OperationsTableRowActions) {
        return child;
      }
    });

  return (
    <div className="dataset-row p-3 border-bottom">
      <div className="col-md-12">
        <DatasetRowItem text="Hola" addClass="dataset-row-title h4">
          <span className="badge badge-warning align-middle">Draft</span>
        </DatasetRowItem>
        <DatasetRowItem addClass="mb-2" text="Lets rock people we got this"></DatasetRowItem>

        <DatasetRowItem addClass="dataset-row-actions mb-1">{renderActions()}</DatasetRowItem>

        <DatasetRowItem addClass="h6 dataset-row-footer" text="Updated 2 days ago by">
          <span>Edwin P. Magezi</span> from <span className="text-uppercase text-danger">yolo</span>
        </DatasetRowItem>
      </div>
    </div>
  );
};
