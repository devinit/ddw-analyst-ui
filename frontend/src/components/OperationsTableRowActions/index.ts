import { OperationsTableRowActions } from './OperationsTableRowActions';

export * from './OperationsTableRowActions';
export default OperationsTableRowActions;
