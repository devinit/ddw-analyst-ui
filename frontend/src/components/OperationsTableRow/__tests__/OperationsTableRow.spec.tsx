/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';
import { List, Map } from 'immutable';
import 'jest-styled-components';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Operation, OperationMap } from '../../../types/operations';
import OperationsTableRowActions from '../../OperationsTableRowActions';
import { OperationsTableRow, OperationsTableRowProps } from '../OperationsTableRow';

const operation: Partial<Operation> = {
  name: 'My Test Dataset',
  description: 'My Dataset Description',
  updated_on: new Date('2020-01-02').toISOString(),
  is_draft: false,
  operation_steps: List(),
};
const props: OperationsTableRowProps = {
  operation: Map(operation) as OperationMap,
};

describe('OperationsTableRow', () => {
  test('renders correctly with the default props', () => {
    const renderer = TestRenderer.create(<OperationsTableRow {...props} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders actions when provided', () => {
    const renderer = TestRenderer.create(
      <OperationsTableRow {...props}>
        <OperationsTableRowActions>
          <button type="button">Click Me</button>
        </OperationsTableRowActions>
      </OperationsTableRow>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('does not render non-action children', () => {
    const renderer = TestRenderer.create(
      <OperationsTableRow {...props}>
        <button type="button">Click Me</button>
      </OperationsTableRow>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('that is a configured to show a draft badge renders a draft badge of warning variant for draft operations', () => {
    const { getByTestId } = render(
      <OperationsTableRow
        operation={Map({ ...operation, is_draft: true }) as OperationMap}
        showDraftBadge
      />,
    );
    const badge = getByTestId('draft-span');

    expect(badge.classList).toContain('badge-warning');
    expect(badge.innerHTML).toBe('Draft');
  });

  test('that is not a draft does not render a draft badge', () => {
    const { queryByTestId } = render(<OperationsTableRow {...props} showDraftBadge />);
    const badge = queryByTestId('draft-span');

    expect(badge).toBeNull();
  });
});
