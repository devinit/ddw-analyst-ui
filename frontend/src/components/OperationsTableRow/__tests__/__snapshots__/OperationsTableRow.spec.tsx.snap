// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationsTableRow does not render non-action children 1`] = `
.c0 {
  display: block !important;
}

<tr
  data-testid="operations-table-row"
>
  <td>
    My Test Dataset
  </td>
  <td>
    Wed Jan 01 2020
  </td>
  <td>
    <span
      className="badge badge-danger"
      data-testid="op-table-row-badge"
    >
      Published
    </span>
  </td>
  <td
    className="c0 td-actions text-right"
  />
</tr>
`;

exports[`OperationsTableRow renders actions when provided 1`] = `
.c0 {
  display: block !important;
}

<tr
  data-testid="operations-table-row"
>
  <td>
    My Test Dataset
  </td>
  <td>
    Wed Jan 01 2020
  </td>
  <td>
    <span
      className="badge badge-danger"
      data-testid="op-table-row-badge"
    >
      Published
    </span>
  </td>
  <td
    className="c0 td-actions text-right"
  >
    <button
      type="button"
    >
      Click Me
    </button>
  </td>
</tr>
`;

exports[`OperationsTableRow renders correctly with the default props 1`] = `
.c0 {
  display: block !important;
}

<tr
  data-testid="operations-table-row"
>
  <td>
    My Test Dataset
  </td>
  <td>
    Wed Jan 01 2020
  </td>
  <td>
    <span
      className="badge badge-danger"
      data-testid="op-table-row-badge"
    >
      Published
    </span>
  </td>
  <td
    className="c0 td-actions text-right"
  />
</tr>
`;
